src/mocks/MockRouterV2.sol

```diff
@@ -599,97 +563,31 @@ contract MockRouterV2 is
-        require(
-            block.timestamp < scheduledTimestampForUpgrade,
-            ErrorsLib.TooLateToCancelUpgrade(scheduledTimestampForUpgrade)
-        );
-
-        string memory action = "cancel";
-        (, bytes memory messageAsG1Bytes,) = contractUpgradeParamsToBytes(
-            action, scheduledImplementation, scheduledImplementationCalldata, scheduledTimestampForUpgrade
-        );
-
-        require(
-            contractUpgradeBlsValidator.verifySignature(
-                messageAsG1Bytes, signature, contractUpgradeBlsValidator.getPublicKeyBytes()
-            ),
-            ErrorsLib.BLSSignatureVerificationFailed()
-        );
-
-        address cancelledImplementation = scheduledImplementation;
-        scheduledImplementation = address(0);
-        scheduledTimestampForUpgrade = 0;
-        scheduledImplementationCalldata = "";
-        emit UpgradeCancelled(cancelledImplementation);
+    function cancelUpgrade(bytes calldata signature) public override (IRouter, ScheduledUpgradeable) {
+        super.cancelUpgrade(signature);
     }
 
     /// @notice Executes a scheduled upgrade
-    function executeUpgrade() external {
-        require(scheduledImplementation != address(0), ErrorsLib.NoUpgradePending());
-        require(
-            block.timestamp >= scheduledTimestampForUpgrade, ErrorsLib.UpgradeTooEarly(scheduledTimestampForUpgrade)
-        );
-
-        address impl = scheduledImplementation;
-        bytes memory callData = scheduledImplementationCalldata;
-
-        // Reset state BEFORE performing upgrade to avoid reentrancy issues
-        scheduledImplementation = address(0);
-        scheduledTimestampForUpgrade = 0;
-        scheduledImplementationCalldata = "";
-
-        // External call to self for msg.sender == address(this) inside _authorizeUpgrade to be valid
-        (bool success, bytes memory ret) =
-            address(this).call(abi.encodeWithSelector(this.upgradeToAndCall.selector, impl, callData));
-
-        if (!success) {
-            // Bubble up revert reason if present
-            if (ret.length > 0) {
-                assembly {
-                    let size := mload(ret)
-                    revert(add(ret, 32), size)
-                }
-            }
-            revert ErrorsLib.UpgradeFailed();
-        }
-
-        emit UpgradeExecuted(impl);
+    function executeUpgrade() public override (IRouter, ScheduledUpgradeable) {
+        super.executeUpgrade();
     }
 
     // ---------------------- Internal Functions ----------------------
 
     /// @dev Required by UUPS to restrict upgrades.
-    function _authorizeUpgrade(address /* newImplementation */ ) internal view override {
-        // Only allow calls coming from within this contract
-        require(msg.sender == address(this), ErrorsLib.UpgradeMustGoThroughExecuteUpgrade());
+    function _authorizeUpgrade(address newImplementation) internal view override {
+        super._authorizeUpgrade(newImplementation);
```

Comment:
if all the function does is pass through its parent, afaik you can just leave it implicit

Reply:
I've removed this function from the `Router.sol` contract. It's internal as well.